src/unit/engine/UberAndroidTestEngine.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+final class UberAndroidTestEngine extends ArcanistUnitTestEngine {
+
+    public function run() {
+        $results = $this->checkstyle();
+        return $results;
+    }
+
+    private function run_command($command) {
+        exec($command, $output, $return_code);
+
+        $result = new ArcanistUnitTestResult();
+        $result->setName($command);
+        $result->setResult($return_code == 0 ? ArcanistUnitTestResult::RESULT_PASS : ArcanistUnitTestResult::RESULT_FAIL);
+
+        return array($result);
+    }
+
+    private function checkstyle() {
+        return $this->run_command('./gradlew checkstyleAll');
```

Comment:
You should check if the binary is installed, like this:

https://github.com/uber/arcanist/blob/63e6d3918a2c3ac4d92b7eb7f9568e480d7a7a0a/src/lint/linter/ArcanistGoVetLinter.php#L26-L47